### PR TITLE
IGNITE-21521 Fix update order in upsertAll and DataStreamer for a new key

### PR DESCRIPTION
**Pull request by ptupitsyn**
Use the same `RowId` when the same key occurs multiple times within a batch in `RW_UPSERT_ALL` handler in `PartitionReplicaListener` to apply the last update for a given key, not a random one.

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java [hunk 1]
```diff
@@ -2250,9 +2256,12 @@ private CompletableFuture<ReplicaResult> processMultiEntryAction(ReadWriteMultiR
 
                         boolean insert = rowId == null;
 
-                        RowId rowId0 = insert ? new RowId(partId(), UUID.randomUUID()) : rowId;
+                        RowId rowId0 = insert
```
Comment (ascherbakoff) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java:2259]: Can we avoid taking locks for duplicate rows (in line 2275) ?

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java [hunk 2]
```diff
@@ -2250,9 +2256,12 @@ private CompletableFuture<ReplicaResult> processMultiEntryAction(ReadWriteMultiR
...
+                                ? newKeyMap.computeIfAbsent(pk.byteBuffer(), rowIdGen)
+                                : rowId;
 
                         if (lastCommitTime != null) {
+                            assert rowId != null : "lastCommitTime is not null, but rowId is null";
```
Comment (ascherbakoff) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java:2264]: This assert looks unrelated to bug. This is executed only for existing rows.

Comment (ptupitsyn) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java]: It's just to fix the IDE warning. Replaced it with `noinspection`.

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java (hunk 1 above)
Comment (ptupitsyn) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java]: Good catch. Fixed to reuse the lock and skip previous updates for the same key.

**Review by ascherbakoff — APPROVED** [review]
